Feature/news filters active 218

lukewhitmore opened the pull request:
Added active class to filter items, based on value in URL parameter.

https://github.com/wellcometrust/upd/issues/218

alfraser86 commented on www/themes/upd/upd.theme:
```diff
@@ -213,15 +213,29 @@ function upd_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+
+      // If term is selected, set active class.
+      if ($k == $field_news_category_target_id) {
+        $state_class = ' active';
```
Maybe use a ternary for readability?
````
$state_class = $k == $field_news_category_target_id ? 'active' : '';

alfraser86 commented on www/themes/upd/upd.theme:
```diff
@@ -213,15 +213,29 @@ function upd_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+
       if ($k == "All") {
-        $html .= '<a href="#" class="news-cat-changer active" data-term-id="' . $k . '">All</a>';
+        $html .= '<a href="#" class="news-cat-changer' . $state_class . '" data-term-id="' . $k . '">All</a>';
```
No L function for D8? always feel it's better than concatenating string.

lukewhitmore commented:
Hey @alfraser86, I've refactored this to use D8 conventions, also reworked it so the solution works without JS (previously would only work with JS on).

alfraser86 commented on www/themes/upd/upd.theme:
```diff
@@ -213,16 +215,52 @@ function upd_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+      if ($article_count > 0 || $k == 'All') {
+
+        // If term is selected, set active class.
+        if ($k == $field_news_category_target_id) {
```
ternary :-)

lukewhitmore replied:
Gah, sorry .. fixed!

lukewhitmore replied:
In my defence; Drupal coding standards :) .. *ducks*
